tests/unit/docs/test_service.py

```diff
@@ -0,0 +1,127 @@
+        contents = service_documenter.document_service().decode('utf-8')
+        lines = [
+            '*********',
+            'MyService',
+            '*********',
+            '.. contents:: Table of Contents',
+            '   :depth: 2',
+            '======',
+            'Client',
+            '======',
+            '.. py:class:: MyService.Client',
+            '  These are the available methods:',
+            '  *   :py:meth:`sample_operation`',
+            '==========',
+            'Paginators',
+            '==========',
+            'The available paginators are:',
+            '* :py:class:`MyService.Paginator.sample_operation`',
+            '.. py:class:: MyService.Paginator.sample_operation',
+            '  .. py:method:: paginate(**kwargs)',
+            '=======',
+            'Waiters',
+            '=======',
+            'The available waiters are:',
+            '* :py:class:`MyService.Waiter.sample_operation_complete`',
+            '.. py:class:: MyService.Waiter.sample_operation_complete',
+            '  .. py:method:: wait(**kwargs)',
+            '================',
+            'Service Resource',
+            '================',
+            '.. py:class:: MyService.ServiceResource()',
+            "  These are the resource's available actions:",
+            '  *   :py:meth:`sample_operation()`',
+            "  These are the resource's available sub-resources:",
+            '  *   :py:meth:`Sample()`',
+            "  These are the resource's available collections:",
+            '  *   :py:attr:`samples`',
+            '  .. py:method:: sample_operation(**kwargs)',
+            '  .. py:method:: Sample(name)',
+            '  .. py:attribute:: samples',
+            '    .. py:method:: all()',
+            '    .. py:method:: filter(**kwargs)',
+            '    .. py:method:: limit(**kwargs)',
+            '    .. py:method:: page_size(**kwargs)',
+            '======',
+            'Sample',
+            '======',
+            '.. py:class:: MyService.Sample(name)',
+            "  These are the resource's available identifiers:",
+            '  *   :py:attr:`name`',
+            "  These are the resource's available attributes:",
+            '  *   :py:attr:`foo`',
+            '  *   :py:attr:`bar`',
+            "  These are the resource's available actions:",
+            '  *   :py:meth:`load()`',
+            '  *   :py:meth:`operate()`',
+            '  *   :py:meth:`reload()`',
+            "  These are the resource's available waiters:",
+            '  *   :py:meth:`wait_until_complete()`',
+            '  .. py:attribute:: name',
+            '  .. py:attribute:: foo',
+            '  .. py:attribute:: bar',
+            '  .. py:method:: load()',
+            '  .. py:method:: operate(**kwargs)',
+            '  .. py:method:: reload()',
+            '  .. py:method:: wait_until_complete(**kwargs)',
+        ]
+        for line in lines:
```

Comment:
Up to you.  It just stood out because it was the only test not using it.  Maybe a comment explaining the situation in case someone decides to add it in the future.